conans/client/graph/graph_binaries.py

```diff
@@ -102,15 +118,32 @@ def _evaluate_node(self, node, build_mode, update, evaluated_references, remote_
                 node.binary = BINARY_CACHE
                 package_hash = ConanInfo.load_from_package(package_folder).recipe_hash
         else:  # Binary does NOT exist locally
+
+            if revisions_enabled:
+                # If revisions enabled, even if the "remote" has no the binary search it in the
+                # rest of remotes. It will look for the binary for the same recipe revision
+                iterate_remotes = [r for r in remotes if r != remote] if remotes else []
+            else:
+                # Do not search for packages for the specific resolved recipe revision but all
+                package_ref = package_ref.copy_clear_rev()
+                if remote:
+                    # Search only in the current remote, do not iterate
+                    iterate_remotes = []
+                else:
+                    # No revisions but no current remote, search everywhere
+                    iterate_remotes = remotes
+
             remote_info = None
             if remote:
                 remote_info = self._get_package_info(package_ref, remote)
-            elif remotes:  # Iterate all remotes to get this binary
-                for r in remotes:
+
+            if not remote_info:
```

Comment:
Not sure, so this means than when revisions are fully enabled, the remotes are always iterated looking for binaries, even if ``-r=myremote`` is specified?

Reply:
Fixed. Only iterating if the user didn't specified the `-r` (but iterating even if a registry entry exists)